lib/Sema/CSDiagnostics.cpp

```diff
@@ -455,8 +455,15 @@ bool RequirementFailure::diagnoseAsError() {
 
   if (auto *OTD = dyn_cast<OpaqueTypeDecl>(AffectedDecl)) {
     auto *namingDecl = OTD->getNamingDecl();
-    emitDiagnostic(diag::type_does_not_conform_in_opaque_return,
-                   namingDecl, lhs, rhs, rhs->isAnyObject());
+
+    auto &req = getRequirement();
+    if (req.getKind() == RequirementKind::SameType) {
+      emitDiagnostic(diag::type_is_not_equal_in_opaque_return, namingDecl, lhs,
+                     rhs);
+    } else {
```

Comment:
It seems we diagnose layout requirements as conformance to AnyObject elsewhere so that’s ok, but maybe convert this to a switch and assert that there’s no SameShape

Reply:
Will do!

Reply:
✅